src/prefect/environments/storage/docker.py

```diff
@@ -464,7 +464,7 @@ def create_dockerfile_object(self, directory: str) -> str:
         if self.env_vars:
             white_space = " " * 20
             env_vars = "ENV " + " \\ \n{}".format(white_space).join(
-                "{k}={v}".format(k=k, v=v) for k, v in self.env_vars.items()
+                '{k}="{v}"'.format(k=k, v=v) for k, v in self.env_vars.items()
```

Comment:
This won't work if the environment variable has double quotes in it already. Since docker accepts both single or double quotes, we should be able to rely on repr here instead:

```suggestion
                f"{k}={v!r}" for k, v in self.env_vars.items()
```

Reply:
Ooh that's fun, I'll add coverage for this case as well.

Reply:
Hm that gives `STR_WITH_QUOTES="Hello "friend""`

Reply:
We may be best off just dropping this and relying on the user to quote their variables as needed

Reply:
? I'm unable to reproduce:

```python
In [39]: v = '"has quotes"'

In [40]: k = 'FOO'

In [41]: print(f"{k}={v!r}")
FOO='"has quotes"'
```

Reply:
You're right that is working as expected, there are some other wonky things going on with the dockerfile formatting that I'm digging into now